src/libs/events.js

```diff
@@ -63,4 +65,48 @@ export const PageViewReporter = () => {
   return null
 }
 
+export const captureAppcuesEvent = (eventName, event) => {
+  // Only record "public-facing events" (and related properties) as documented by Appcues: https://docs.appcues.com/article/301-client-side-events-reference
+  const publicEvents = [
+    'flow_started',
+    'flow_completed',
+    'flow_skipped',
+    'flow_aborted',
+    'step_started',
+    'step_completed',
+    'step_skipped',
+    'step_aborted',
+    'step_interacted',
+    'form_submitted',
+    'form_field_submitted'
+  ]
+  if (_.includes(eventName, publicEvents)) {
+    const eventProps = {
+      'appcues.flowId': event.flowId,
+      'appcues.flowName': event.flowName,
+      'appcues.flowType': event.flowType,
+      'appcues.flowVersion': event.flowVersion,
+      'appcues.id': event.id,
+      'appcues.interaction.category': event.interaction?.category,
+      'appcues.interaction.destination': event.interaction?.destination,
+      'appcues.interaction.element': event.interaction?.element,
+      'appcues.interaction.fields': event.interaction?.fields ? JSON.stringify(event.interaction.fields) : undefined,
```

Comment:
Still (re-)new to JS. The linter over in Martha has yelled at me for using the `undefined` literal before. Would this line be the same if written as:

```javascript
'appcues.interaction.fields': event.interaction?.fields && JSON.stringify(event.interaction.fields),
```

Reply:
I considered that, but I was afraid of it evaluating to a boolean.

Reply:
We're fine with undefined as a literal in this codebase, I'm not sure why Martha wouldn't be.

If you try it in the browser console, though, you'll find that, due to short-circuiting, `undefined && true` is `undefined`, rather than `false`, so you should be good with @kshakir's syntax.

Reply:
Here's the linter's explanation: https://eslint.org/docs/rules/no-undefined